src/pages/register-a-beacon/application-complete.tsx

```diff
@@ -74,28 +76,46 @@ const ApplicationCompleteWhatNext: FunctionComponent = (): JSX.Element => (
 export const getServerSideProps: GetServerSideProps = withCookieRedirect(
   async (context: GetServerSidePropsContext) => {
     const decoratedContext = await decorateGetServerSidePropsContext(context);
+    const registrationClass = decoratedContext.registration;
     const registration = decoratedContext.registration.registration;
 
     let pageSubHeading;
 
-    if (!registration.reference) {
-      registration.reference = referenceNumber("A#", 7);
-
-      const govNotifyGateway = new GovNotifyGateway();
-      const sendGovNotifyEmailUseCase = new SendGovNotifyEmail(
-        govNotifyGateway
+    if (!registration.referenceNumber) {
+      registration.referenceNumber = referenceNumber("A#", 7);
+      const beaconsApiGateway = new BeaconsApiGateway();
+      const createRegistrationUseCase = new CreateRegistration(
+        beaconsApiGateway
+      );
+      const success = await createRegistrationUseCase.execute(
+        registrationClass
       );
-      if (sendGovNotifyEmailUseCase.execute(registration)) {
-        pageSubHeading = "We have sent you a confirmation email - ";
+
+      if (success) {
+        const govNotifyGateway = new GovNotifyGateway();
+        const sendGovNotifyEmailUseCase = new SendGovNotifyEmail(
+          govNotifyGateway
+        );
+
+        const emailSuccess = await sendGovNotifyEmailUseCase.execute(
+          registration
+        );
+
+        if (emailSuccess) {
+          pageSubHeading = "We have sent you a confirmation email.";
+        } else {
+          pageSubHeading =
+            "We could not send you a confirmation email. But we have registered your beacon under the following reference id.";
+        }
       } else {
-        pageSubHeading = "We could not send you a confirmation email.";
+        delete registration.referenceNumber;
+        pageSubHeading =
+          "We could not save your registration or send you a confirmation email. Please contact the Beacons Registry team.";
```

Comment:
Unsure how we best handle this? The impact of this, i.e: a user not able to register a beacon, would be reduced if we moved the caching to the backend